app/lib/dartdoc/backend.dart

```diff
@@ -320,3 +320,58 @@ class DartdocBackend {
     await deletePool.close();
   }
 }
+
+class ArchiveTracker {
+  final _pvMap = <String, _TrackedEntry>{};
+  final _sampleLines = <String>[];
+
+  ArchiveTracker() {
+    _logger.info('Tracking: init.');
+    new Timer.periodic(const Duration(minutes: 10), (_) => _cleanup());
+    new Timer.periodic(const Duration(minutes: 45), (_) => _logSample());
+  }
+
+  List<String> get samples => _sampleLines;
+
+  void track(DartdocEntry entry) {
+    final key = '${entry.packageName}/${entry.packageVersion}';
+    final tracked = _pvMap.putIfAbsent(key, () => new _TrackedEntry());
+    tracked.archiveSize = entry.archiveSize ?? 0;
+    tracked.totalSize = entry.totalSize ?? 0;
+    tracked.lastAccess = new DateTime.now();
+  }
+
+  void _cleanup() {
+    final now = new DateTime.now();
+    _pvMap.removeWhere((key, value) {
+      final age = value.lastAccess.difference(now).abs();
```

Comment:
No need for abs() if you use monotonic time (where you know now is not before lastAccess) and you do "now.difference(lastAccess)".

Reply:
Fixing. Bad habit: I don't want to think about which should be the later timestamp :)

Reply:
Yeah I always look at up as well.